src/Geno/C++/Components/Workspace.cpp

```diff
@@ -138,41 +171,26 @@ void Workspace::Rename( std::string Name )
 
 //////////////////////////////////////////////////////////////////////////
 
-Project& Workspace::NewProject( std::filesystem::path Location, std::string Name )
+Project* Workspace::NewProject( std::filesystem::path Location, std::string Name )
 {
-	Project& project = m_Projects.emplace_back( std::move( Location ) );
-	project.m_Name   = std::move( Name );
+	Project* pProject = new Project( std::move( Location ), std::move( Name ) );
+	AddChild( pProject );
 
-	return project;
+	return pProject;
```

Comment:
Returning a pointer indicates that the function may fail, which it can't. Make it still return a reference to a Project. Just dereference the pointer.